src/adapter/src/coord.rs

```diff
@@ -845,6 +846,16 @@ impl PlanValidity {
                 )));
             }
         }
+
+        if let Some(create_sql) = &self.create_sql {
+            let wall_time = to_datetime((catalog.config().now)());
+            let pcx = PlanContext::new(wall_time);
+            let conn_catalog = catalog.state().for_session(session);
+            if let Err(e) = CatalogState::parse_plan(create_sql, Some(&pcx), &conn_catalog) {
+                return Err(AdapterError::ChangedPlan(e.to_string()));
```

Comment:
I'm not convinced this is the right error. I think it should be a 40001 which signals to the client that they should re-run their statement. This one is about DECLARE/FETCH stuff usually.